### PR TITLE
Add auto-detection of AWS signing scope for managed OpenSearch connections

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary

Previously the AWS OpenSearch client was hardcoded to use the `aoss` signing scope (OpenSearch Serverless). This caused failures when connecting to managed (non-serverless) OpenSearch instances, which require a different scope (e.g., `es`).

This change adds automatic scope detection: the client first attempts a connection using `aoss`, and if AWS returns a "Credential should be scoped to correct service" error, it extracts the correct scope from the error message and retries. This allows the registry loader to work with both serverless and managed OpenSearch without manual configuration.

### 🤖 AI Assistance Disclosure
- [x] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: _0_ %

## ⚙️ Test Data and/or Report
Running by hand to operate on a managed opensearch instance.

## ♻️ Related Issues
Part 1 of NASA-PDS/registry#524

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java [hunk 1]
```diff
@@ -96,15 +97,29 @@ public RestClientWrapper(ConnectionFactory conFact, boolean isServerless) {
...
+                this.httpClient,
+                this.conFact.getHostName(),
+                this.scope,
+                Region.US_WEST_2, // signing service region that we should probably get from host name??
+                AwsSdk2TransportOptions.builder().build()
+                )
+            );
+        client.info();
+      } catch (IOException | OpenSearchException e) {
+        if (e.getMessage().contains("Credential should be scoped to correct service:")) {
```
Comment (nutjob4life) [inline common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java]: Possible `NullPointerException`. Consider:
```java
String message = e.getMessage();
if (message != null && message.contains(...)) {
```

File: common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java [hunk 2]
```diff
@@ -96,15 +97,29 @@ public RestClientWrapper(ConnectionFactory conFact, boolean isServerless) {
...
+          int first = e.getMessage().indexOf("'") + 1;
+          int last = e.getMessage().lastIndexOf("'");
```
Comment (nutjob4life) [inline common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java:113]: Maybe add a sanity check just in case a message is particularly "garbled"?

    first > 0 && last > first

Or not?

File: common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java [hunk 3]
```diff
@@ -96,15 +97,29 @@ public RestClientWrapper(ConnectionFactory conFact, boolean isServerless) {
...
+          String scop = e.getMessage().substring(first, last);
```
Comment (nutjob4life) [inline common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java]: SCOP? Okay this works but … scop? 🤭

### It's the SCOP!

# 🆂👮

**Review by nutjob4life — APPROVED** [review]
Code delta: ✓ (see 3 minor interspersed comments)
AI-assisted review: ✓
Tests: ✓
Approval: ✅
Test details:
```
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for gov.nasa.pds:registry-loader 1.4.0-SNAPSHOT:
[INFO] 
[INFO] gov.nasa.pds:registry-loader ....................... SUCCESS [  0.230 s]
[INFO] PDS Registry Common ................................ SUCCESS [ 19.433 s]
[INFO] Harvest Tool ....................................... SUCCESS [  4.050 s]
[INFO] PDS Registry Manager ............................... SUCCESS [  4.709 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

**Comment by jordanpadams** [comment]
@al-niessner ☝️

**Comment by jordanpadams** [comment]
@al-niessner do you mind poking through some of the comments @nutjob4life provided above when you have chance? Or I can sent copilot off to try to do it

File: common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java (hunk 1 above)
Comment (al-niessner) [reply common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java]: fixed

File: common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java (hunk 2 above)
Comment (al-niessner) [reply common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java:113]: fixed

File: common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java (hunk 3 above)
Comment (al-niessner) [reply common/src/main/java/gov/nasa/pds/registry/common/connection/aws/RestClientWrapper.java]: fixed, but do not complain about name overlap.
